src/main.cpp

```diff
@@ -0,0 +1,40 @@
+#include <stdio.h>
+
+#include "bsp/Factory.h"
+#include <FreeRTOS.h>
+#include <FreeRTOSConfig.h>
+#include <task.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+void dummyTask(void* param) {
+    (void)param;
+    printf("Hello world!\n");
+
+    /* Print chip information */
+    std::shared_ptr<IBSP> bsp = BspFactory::getBSP();
+    ChipInfo info = bsp->getChipInfo();
+
+    while (true) {
+        printf("System has %d cores\n\r", info.m_cores);
+        if (info.m_osType == ChipInfo::ESP) {
+            printf("System is running on target\n\r");
+        } else {
+            printf("System is running locally\n\r");
+        }
+        vTaskDelay(1000 / portTICK_PERIOD_MS);
+    }
+}
+
+void app_main(void) {
```

Comment:
Why app_main? is it because the ESP uses it instead of main? If so you could probably use a interface linker library link in the hivemind for the  interrupts.

Reply:
Not sure what you mean. This will be the main loop for both build. Interrupts are only relevant for the stm build on the HiveMind. You can check System.cpp in the posix bsp to see how this function called.

Reply:
I mean the other way around, so your "System.cpp" would be in the esp bsp with an app_main that calls the main function.

Instead of a "system.cpp" in the posix bsp that calls the app_main.